lib/store_web/live/product_live/show.ex

```diff
@@ -15,4 +22,72 @@ defmodule StoreWeb.ProductLive.Show do
      |> assign(:current_page, :products)
      |> assign(:product, Inventory.get_product!(id))}
   end
+
+  def redeem_quantity(user_id, product_id) do
+    order_quantity = Enum.count(Inventory.list_orders(where: [user_id: user_id]))
+
+    quantity =
+      case order_quantity do
+        0 -> Inventory.get_product!(product_id).max_per_user
+        _ -> Inventory.get_product!(product_id).max_per_user - order_quantity
+      end
+
+    if quantity < 0 do
+      0
+    else
+      quantity
+    end
+  end
```

Comment:
You should only be able to redeem from one order at a time. Also, if quantity is less than 0, you should raise as error (as that means there was a bug somewhere), instead of letting it pass undetected